codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AwsJsonRpc1_0.java

```diff
@@ -36,4 +45,77 @@ protected String getDocumentContentType() {
+        GenerationContext context, OperationShape operation, Symbol responseType) {
+        if (isAwsQueryCompatibleTraitFound(context)) {
+            return HttpProtocolGeneratorUtils.generateErrorDispatcher(
+                context, operation, responseType, this::writeErrorMessageCodeDeserializer,
+                this::getOperationErrors, (writer) -> AwsJsonRpc1_0.awsQueryCompatibleDefaultBlockWriter(writer));
+        } else {
+            return HttpProtocolGeneratorUtils.generateErrorDispatcher(
+                context, operation, responseType, this::writeErrorMessageCodeDeserializer,
+                this::getOperationErrors);
+        }
+    }
+
+    @Override
+    protected void deserializeError(GenerationContext context, StructureShape shape) {
+        GoWriter writer = context.getWriter().get();
+        Symbol symbol = context.getSymbolProvider().toSymbol(shape);
+        String functionName = ProtocolGenerator.getDocumentDeserializerFunctionName(shape, context.getService(),
+                getProtocolName());
+
+        AwsProtocolUtils.initializeJsonDecoder(writer, "errorBody");
+        AwsProtocolUtils.decodeJsonIntoInterface(writer, "");
+        writer.write("output := &$T{}", symbol);
+        writer.write("err := $L(&output, shape)", functionName);
+        writer.write("");
+        AwsProtocolUtils.handleDecodeError(writer);
+        writer.write("errorBody.Seek(0, io.SeekStart)");
+        if (isAwsQueryCompatibleTraitFound(context)) {
+            writer.write("awsQueryErrorCode := getAwsQueryErrorCode(response)");
+            writer.openBlock("if awsQueryErrorCode != \"\" {", "}", () -> {
+                writer.write("output.ErrorCodeOverride = &awsQueryErrorCode");
+            });
+        }
+        writer.write("return output");
+    }
+
+    @Override
+    public void generateSharedDeserializerComponents(GenerationContext context) {
+        super.generateSharedDeserializerComponents(context);
+        if (isAwsQueryCompatibleTraitFound(context)) {
+            GoWriter writer = context.getWriter().get();
+            writer.openBlock("func getAwsQueryErrorCode(response *smithyhttp.Response) string {", "}", () -> {
+                writer.write("queryCodeHeader := response.Header.Get(\"x-amzn-query-error\")");
+                writer.openBlock("if queryCodeHeader != \"\" {", "}", () -> {
+                    writer.write("queryCodeParts := strings.Split(queryCodeHeader, \";\")");
+                    writer.openBlock("if queryCodeParts != nil && len(queryCodeParts) == 2 {", "}", () -> {
+                        writer.write("return queryCodeParts[0]");
+                    });
+                });
+                writer.write("return \"\"");
+            });
+        }
+    }
+
+    private static void awsQueryCompatibleDefaultBlockWriter(GoWriter writer) {
+        writer.openBlock("default:", "", () -> {
+            writer.write("awsQueryErrorCode := getAwsQueryErrorCode(response)");
```

Comment:
Yes, we can use a text block here!

We'd also want to verify the code written still parses correctly since the text block will not be aware of the writer level indentation (could be overthinking it since switch statement clauses have implicit blocks even without curly braces).

A text block:

```java
writer.write("""
    default:
        awsQueryErrorCode := getAwsQueryErrorCode(response)
        if awsQueryErrorCode != ""  {
            errorCode = awsQueryErrorCode
        }
        genericError := &smithy.GenericAPIError{
            Code: errorCode,
            Message: errorMessage,
        }
        return genericError""");
```

will produce something like this:

```go
    switch {
// defaultBlockWriter start
// indentation from writer indentation level
        default:
// following lines are not aware of writer indentation level,
// only text block indentation level
    awsQueryErrorCode := getAwsQueryErrorCode(response)
    if awsQueryErrorCode != ""  {
        errorCode = awsQueryErrorCode
    }
    genericError := &smithy.GenericAPIError{
        Code: errorCode,
        Message: errorMessage,
    }
    return genericError
// defaultBlockWriter end
    }
```

which hopefully after `go fmt` / other SDK release tools becomes:

```go
    switch {
    default:
        awsQueryErrorCode := getAwsQueryErrorCode(response)
        if awsQueryErrorCode != "" {
            errorCode = awsQueryErrorCode
        }
        genericError := &smithy.GenericAPIError{
            Code:    errorCode,
            Message: errorMessage,
        }
        return genericError
    }
```